app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,173 @@
+    }
+
+    public void onClickByCity(View view) {
+        Log.d("myTag", "OnClick start");
+        String city = msgCity.getText().toString();
+        if (!checkCityString(city)) {
+            msgOnWrongCity();
+            return;
+        }
+
+        String request = formatRequestWithCity(city);
+        Log.d("myTag", request);
+        String response = doRequest(request);
+        if (response != null) {
+            Log.d("myTag", "len:" + response.length());
+        } else {
+            msgOnWrongRequest();
+            return;
+        }
+
+        String formattedOutput = getTemperature(response);
+        if (formattedOutput == null) {
+            msgOnWrongCity();
+            return;
+        }
+        String newMessage = getText(R.string.default_message) + " " + formattedOutput + " \u00B0C";
+        msgTemperature.setText(newMessage);
+        Log.d("myTag", "OnClick finish");
+    }
+
+    public void onClickByLocation(View view) {
+
+    }
+
+    private boolean checkCityString(String city) {
+        if (city == null) {
+            msgOnWrongCity();
+            return false;
+        }
+        else if (city.length() < 2 || city.length() > 30) {
+            msgOnWrongCity();
+            return false;
+        }
+        // TODO: do some other checking of input string
+        return true;
+    }
+
+    private String formatRequestWithCity(String city) {
+        String api_url = getText(R.string.weather_entry_point).toString();
+        String key = getText(R.string.weather_key).toString();
+        return api_url + "?q=" + city + "&appid=" + key + "&units=metric";
+    }
+
+    private String getTemperature(String response) {
+        String ret = null;
+
+        try {
+            JSONObject json = new JSONObject(response);
+            JSONObject main  = json.getJSONObject("main");
+            ret = main.getString("temp");
+            Log.d("myTag", "temperature = " + ret);
+        } catch (JSONException e) {
+            msgOnWrongRequest();
+            e.printStackTrace();
+        }
+
+        return ret;
+    }
+
+    private String doRequest(String request) {
+        final String[] respond = {null};
+
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                BufferedReader reader = null;
+                InputStream stream = null;
+                HttpsURLConnection connection = null;
+                try {
+                    URL weatherEndpoint = new URL(request);
+                    connection = (HttpsURLConnection) weatherEndpoint.openConnection();
+                    connection.setRequestProperty("User-Agent", getText(R.string.app_name).toString());
+                    connection.setRequestMethod("GET");
+                    connection.setReadTimeout(3000);
+                    connection.connect();
+
+                    if (connection.getResponseCode() == 200) {
```

Comment:
another magic number.
```suggestion
                    if (connection.getResponseCode() == HttpsURLConnection.HTTP_OK) {
```

Reply:
done